packages/builder-vite/code-generator-plugin.ts

```diff
@@ -25,16 +25,21 @@ export function codeGeneratorPlugin(options: ExtendedOptions): Plugin {
       // invalidate the whole vite-app.js script on every file change.
       // (this might be a little too aggressive?)
       server.watcher.on('change', (_e) => {
-        const { moduleGraph } = server;
-        const appModule = moduleGraph.getModuleById(virtualFileId);
+        const appModule = server.moduleGraph.getModuleById(virtualFileId);
         if (appModule) {
           server.moduleGraph.invalidateModule(appModule);
         }
-        const storiesModule = moduleGraph.getModuleById(virtualStoriesFile);
+        const storiesModule = server.moduleGraph.getModuleById(virtualStoriesFile);
         if (storiesModule) {
           server.moduleGraph.invalidateModule(storiesModule);
         }
       });
+
+      server.watcher.on('all', (_e, path) => {
+        if (/\.stories\.([tj])sx?$/.test(path) || /.mdx$/.test(path)) {
```

Comment:
should this be limited to `stories|story.mdx`?

Reply:
Any mdx could be shown I thought? Maybe we should re-use the stories regex though

Reply:
I don't think anything other than a stories mdx will be adding new stories to the sidebar though, which I thought was the reason for this fix?

Reply:
And yeah, I think using the stories regex would be best here, if it works.  I guess we wouldn't be able to if they're using the object-style configuration though, right?  Or is there a utility to turn that into a normal glob pattern?

Reply:
> I don't think anything other than a stories mdx will be adding new stories to the sidebar though, which I thought was the reason for this fix?

ah this just clicked. Good point

Reply:
I'll just stick with regex for now to keep it simple